spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -46,16 +47,32 @@
 
 	private final String location;
 
-	public GoogleStorageResource(Storage storage, String location) {
+	private boolean createBlobIfNotExists = true;
+
+	public GoogleStorageResource(Storage storage, String location,
+			boolean createBlobIfNotExists) {
 		Assert.notNull(storage, "Storage object can not be null");
 		this.storage = storage;
 		this.location = location;
+		this.createBlobIfNotExists = createBlobIfNotExists;
+	}
+
+	public GoogleStorageResource(Storage storage, String location) {
+		this(storage, location, true);
+	}
+
+	public boolean isCreateBlobIfNotExists() {
+		return this.createBlobIfNotExists;
+	}
+
+	public void setCreateBlobIfNotExists(boolean createBlobIfNotExists) {
+		this.createBlobIfNotExists = createBlobIfNotExists;
 	}
 
 	@Override
 	public boolean exists() {
 		try {
-			return resolve().exists();
+			return throwExceptionForNullBlob(resolve()).exists();
```

Comment:
I think this violates the contract of the `exists` method. It shouldn't throw an exception if the file doesn't exist.

Reply:
@meltsufin ,

We don't review until the CLA confirmation. The contributor must confirm that any our comments won't insult him/her.

Thanks for understanding.

Reply:
@ChengyuanZhao is a new member on my team. He's covered under the Google CLA. :-)

Reply:
Sure thing, @meltsufin !

Thank you for confirmation!

Anyway I think the Webhook for CLA is something to must have.

Reply:
Maybe it's already on, but wasn't triggered because Chengyuan has a google.com email?

Reply:
I don't think so. Even my PR's to my own projects are marked with the CLA confirmations:

![image](https://user-images.githubusercontent.com/1071590/31280363-1a594c1e-aa7a-11e7-81b6-5c8fcab79e23.png)


That's why I think we just don't have it in this project.

Reply:
I see

Reply:
The issue might be that @ChengyuanZhao 's membership of the GoogleCloudPlatform organization is not visible.
You need to make it visible from your user settings (and join the organization if you haven't yet).

Reply:
@meltsufin, I don't think this changes the behavior though. Originally resolve() returned this.storage.get() directly, and the .get() function's javadoc says it can return null possibly if the blob doesn't exist, which would mean the function would get an NPE if calling resolve().exists(), right? The difference now is that we get an IOException instead of NPE.

Reply:
I think the idea is to `try...catch` that exception and return `false`.
See `AbstractFileResolvingResource` implementation for example:
```
catch (IOException ex) {
	return false;
}
```
in the end.

Reply:
Ah my apologies I said it wrong. Before it would throw an NPE, but now instead it will throw an IOException, which would cause it to return false.